tests/e2e/globals.js

```diff
@@ -91,85 +45,192 @@ async function next(browser) {
+  // parsing parameters
+  // network
+  process.argv.map(arg => {
+    // selected network
+    if (arg.indexOf("--network=") !== -1)
+      network = arg.slice(arg.indexOf("=") + 1)
+    // frontend uri
+    if (arg.indexOf("--fe=") !== -1) feURI = arg.slice(arg.indexOf("=") + 1)
+    // api uri
+    if (arg.indexOf("--api=") !== -1) apiURI = arg.slice(arg.indexOf("=") + 1)
   })
-  // refreshing and saving all needed info of a newly created account
-  await browser.url(browser.launch_url)
-  const tempAcc = await browser.execute(
-    function(networkData) {
-      // saving account info from localStorage
-      let session = window.localStorage.getItem(
-        `session_${networkData.network}`
+  networkData = networks.find(net => net.network == network)
+  if (!networkData) {
+    throw new Error(`Initial data for "${network}" is not set`)
+  }
+  if (!feURI) {
+    throw new Error(`Frontend uri for "${network}" is not set`)
+  }
+  if (!apiURI) {
+    throw new Error(`API uri for "${network}" is not set`)
+  }
+  browser.launch_url = feURI
+  browser.globals.apiURI = apiURI
+  // checking the API for a localhost API
+  if (apiURI.indexOf("//localhost:") !== -1) {
+    apiUp(browser)
+  }
+  // checking if network is local, the API should be local too
+  if (network.indexOf("local-") === 0) {
+    if (apiURI.indexOf("//localhost:") === -1){
+      throw new Error(
+        `Can't test local network "${network}" against nonlocal API`
       )
-      let wallet
-      if (session) {
-        session = JSON.parse(session)
-        wallet = window.localStorage.getItem(
-          `cosmos-wallets-${session.address}`
+    }
+  }
+  // open default page
+  await browser.url(browser.launch_url).then(() => {
+    browser.execute(
+      function(apiURI) {
+        // setting the api to localStorage
+        window.localStorage.setItem("persistentapi", apiURI)
+      },
+      [apiURI]
+    )
+  })
+  return networkData
+}
+
+async function defineNeededValidators(browser, networkData) {
+  // need to store validators, cause they can shuffle during the test
+  await browser.url(browser.launch_url + "/validators", async () => {
+    const validators = await browser.execute(
+      function() {
+        return new Promise(resolve => {
+          let attempts = 5
+          const f = () => {
+            const validatorLIs = document.getElementsByClassName("li-validator")
+            if (validatorLIs.length < 2 && attempts-- > 0) {
+              setTimeout(f, 2000)
+              return false
+            }
+            if (validatorLIs.length < 2) {
+              throw new Error(`No enough validators to check`)
+            }
+            resolve({
+              first: validatorLIs[0].getAttribute("data-name"),
+              second: validatorLIs[1].getAttribute("data-name")
+            })
+          }
+          f()
+        })
+      },
+      [browser, networkData]
+    )
+    browser.globals.validatorOneName = validators.value.first

```

Comment:
this doesn't get deleted on a refresh?

Reply:
No, those variables are global for all tests during test sessions